registry/session.go

```diff
@@ -222,10 +222,10 @@ func (r *Session) GetRemoteTags(registries []string, repository string, token []
 		logrus.Debugf("Got status code %d from %s", res.StatusCode, endpoint)
 		defer res.Body.Close()
 
-		if res.StatusCode != 200 && res.StatusCode != 404 {
-			continue
-		} else if res.StatusCode == 404 {
+		if res.StatusCode == 404 {
 			return nil, fmt.Errorf("Repository not found")
+		} else if res.StatusCode != 200 {
```

Comment:
Ok was just because if later someone would move the second if above the other then mistakes may happens.